mesh/mesh.cpp

```diff
@@ -5250,6 +5257,181 @@ void Mesh::LoadPatchTopo(std::istream &input, Array<int> &edge_to_knot)
 
    FinalizeTopology();
    CheckBdrElementOrientation(); // check and fix boundary element orientation
+
+   // Generate knot 2 edge mapping -- if not specified
```

Comment:
The algorithm for generating this mapping is not very clear from the code. Can you please add more documentation to describe the algorithm at a high level?

Reply:
I have added some documentation